guava/src/com/google/common/escape/UnicodeEscaper.java

```diff
@@ -242,7 +254,7 @@ protected final String escapeSlow(String s, int index) {
    * @return the Unicode code point for the given index or the negated value of the trailing high
    *     surrogate character at the end of the sequence
    */
-  protected static int codePointAt(CharSequence seq, int index, int end) {
+  protected static int codePointAt(CharSequence seq, @IndexFor("#1") int index, @IndexOrHigh("#1") int end) {
```

Comment:
@NhatDinh, oh, now I see. The `index` parameter should be `@IndexFor("#1")` when the method is entered, but then it can be internally increased, so it can become equal to `seq.length()`.
The problem here is that we would like the index to have type `@IndexFor("#1")` when looking at the parameter from outside the method, but `@IndexOrHigh("#1")` when looking at the variable from inside the method.
So to fix the error, I would suggest leaving `@IndexFor("#1")` on the parameter, but creating a new local variable with `@IndexOrHigh("seq")`, initializing it with the value of `index`, and replacing occurences of `index` with this new variable. That way the value of the parameter will not change in the method and the two roles (parameter vs. local variable) will be separated.

The manual (section 29.1.1) says: 
> Sometimes it is acceptable to make a refactoring, such as extracting an expression into a new local variable in order to annotate its type ...

I think we can apply it here.

Reply:
We can talk about this at our next meeting.

Reply:
@panacekcz yes, I already include this problem in our next meeting Javadoc.